### ISSUE TITLE
WinRM Plugin Registration and Modification

### ISSUE THREAD

**Issue by mirichmo** [PowerShell/PowerShell#1193]
# Overview

On Windows, the PowerShell installer will not automatically enable receiving incoming requests. The feature shall be activated as a post-install step that requires Administrator-level authorization (due to the WinRM plugin's System32 placement requirement).
# Changes Required
## Enable-PSRemoting Changes

Cmdlet implementation should be changed to also 
1. Copy pwrshplugin.dll from %ProgramFiles%\PowerShell<versionX>\ (aka $PSHOME) to %windir%\System32\PowerShell<versionX>
2. Create a the necessary plugin configuration file in %windir%\System32\PowerShell<versionX>
3. Register the plugin as a WinRM endpoint (Proposed name = PowerShell.<versionX>)

Get-PSSessionConfiguration should work with these changes.

**Comment by masaeedu** [comment]
Will the cmdlet have options to choose between SSH and WinRM?

**Comment by mirichmo** [comment]
This issue only applies to Windows + WinRM. Adding support for SSH should be handled through a separate issue and would be a feature request. Please go ahead and file an issue for it if you think it would be useful as part of the overall SSH user experience.

**Comment by joeyaiello** [comment]
I believe that most of this logic is already encapsulated in `Install-PowerShellRemoting.ps1`, but we should bring it into `Enable-PSRemoting`.

**Comment by daxian-dbw** [comment]
@mirichmo This issue was closed automatically after merging #2671. If this issue shouldn't be closed due the pending NanoServer scenario, feel free to re-open it.

**Comment by mirichmo** [comment]
The Nano Server Scenario (registering a different instance of PowerShell Core) is covered by the install script and that still exists. I filed #4528 to document this.

**Comment by BatmanAMA** [comment]
It sounds like this is resolved, however on a new install of powershell 6.1.1 (on WS 2016) ``enable-psremoting`` fails to register a powershell.6 (or powershell.6.1.1) endpoint. I need to still use the ``install-powershellremoting.ps1`` script. 
Is this intended? It may be worth clarifying in the [docs](https://github.com/PowerShell/PowerShell-Docs/blob/staging/reference/docs-conceptual/learn/remoting/WSMan-Remoting-in-PowerShell-Core.md) what the psremoting setup procedure is and removing the reference to this issue. Let me know what you're thinking and I'll happily go submit a pull request to the docs repo.

**Comment by benmccallum** [comment]
After installing PS 7.1 and in the installer checking "Enable Remoting" or whatever the option was I couldn't remote. Full details on this [SO post here](https://stackoverflow.com/questions/65131931). 

I found a kind of solution which is giving the windows user I'm trying to connect with Full Control of `C:\Windows\System32\PowerShell\7.1.0`, but I'd obviously like to know if something else has gone wrong and it seems kinda of related to what this installer was doing though I used Enable-PSRemoting in a 7.1 isntance so not sure... 

Any help would be appreciated!
